kahuna/public/js/search/query.js

```diff
@@ -65,18 +66,28 @@ query.controller('SearchQueryCtrl',
         }));
     }
 
-    $scope.$watchCollection(() => ctrl.filter, onValChange(filter => {
-        filter.uploadedBy = filter.uploadedByMe ? ctrl.user.email : undefined;
-        $state.go('search.results', filter);
-    }));
 
-    // we can't user dynamic values in the ng:true-value see:
+    // we can't use dynamic values in the ng:true-value see:
     // https://docs.angularjs.org/error/ngModel/constexpr
     mediaApi.getSession().then(session => {
         ctrl.user = session.user;
-        ctrl.filter.uploadedByMe = ctrl.uploadedBy === ctrl.user.email;
+        ctrl.pseudoFilter.uploadedByMe = ctrl.filter.uploadedBy === ctrl.user.email;
     });
 
+    // Bi-directional sync uploadedByMe <=> uploadedBy(my email)
+    $scope.$watch(() => $stateParams.uploadedBy, onValChange(uploadedBy => {
+        ctrl.pseudoFilter.uploadedByMe = uploadedBy === ctrl.user.email;
+    }));
+    $scope.$watch(() => ctrl.pseudoFilter.uploadedByMe, onValChange(uploadedByMe => {
+        ctrl.filter.uploadedBy = uploadedByMe ? ctrl.user.email : undefined;
```

Comment:
No, this changes filter, and then the block below sets that into the state, which is reflected in the url.